Add some option sliders

Zonz opened the pull request:
5 sliders for users to set at will, for scripters to alter script behavior according to activated sliders.
Both the name and the tooltip can be changed through a lua script.

Testing script:
```
log("Enabling all sliders")

names = {"Yo momma", "is so fat", "that her", "blood type", "is type Ragu"}

for i = 1, 5 do
	setOptionName(i, names[i])
	setOptionDescription(i, "This is Option " .. i .. "'s description, set by a script.\nThis is a new line.")
	setOption(i, true)
end

log("Option 1 is " .. tostring(getOption(1)) .. "!")
```

Zonz committed:
Add OptionSlider class

Zonz committed:
Add slider UI

Zonz committed:
Add slider structure

Zonz committed:
Add slider array

Zonz committed:
Add slider lua functions

Silv3rPRO commented:
While this is a good feature, it looks way too confusing as it is.
New users won't understand what these five sliders are doing here.

Maybe we could find a way to only display the sliders required by a script.

Also, I'm sure we can remove all the code duplication in `MainWindow` using the `sender` of the event, loops or lambdas.

Zonz committed:
Update BotClient.cs

Zonz committed:
Update OptionSlider.cs

Zonz committed:
Update MainWindow.xaml.cs

Zonz commented:
Ask and ye shall receive. Sliders are now hidden on startup and when loading a script, and are only shown when a property is changed by a lua script.

Consolidated that mess of functions too.

Zonz committed:
Update MainWindow.xaml.cs

Switch all sliders to false on loading a new script

Zonz committed:
Update MainWindow.xaml.cs

Zonz committed:
Update MainWindow.xaml.cs

Zonz committed:
Update MainWindow.xaml.cs

Zonz committed:
Add Reset function

This was necessary to prevent the bot from freezing under certain circumstances.

Zonz committed:
Reset name and description

Zonz committed:
Reset sliders

Zonz committed:
Further consolidation

Zonz committed:
Removed typo

Zonz committed:
More consolidation

Zonz committed:
Update MainWindow.xaml

Zonz committed:
Update MainWindow.xaml.cs

Silv3rPRO commented:
Squashed and merged in https://github.com/Silv3rPRO/proshine/commit/5cc4f7753ea7006218bfe77b8f5f7f1cea646a9f, thanks!
